src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs

```diff
@@ -118,5 +126,53 @@ public static TokenValidationParameters CreateTokenValidationParameters()
 
             return result;
         }
+
+        private static string IssuerValidator(string issuer, SecurityToken securityToken, TokenValidationParameters validationParameters)
+        {
+            if (!validationParameters.ValidIssuers.Any(p => string.Equals(issuer, p, StringComparison.OrdinalIgnoreCase)))
+            {
+                throw new SecurityTokenInvalidIssuerException("IDX10205: Issuer validation failed.")
+                {
+                    InvalidIssuer = issuer,
+                };
+            }
+
+            return issuer;
+        }
+
+        private static bool AudienceValidator(IEnumerable<string> audiences, SecurityToken securityToken, TokenValidationParameters validationParameters)
+        {
+            foreach (string audience in audiences)
+            {
+                if (validationParameters.ValidAudiences.Any(p => string.Equals(audience, p, StringComparison.OrdinalIgnoreCase)))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private static void LogAuthenticationFailure(AuthenticationFailedContext context)
+        {
+            var loggerFactory = context.HttpContext.RequestServices.GetRequiredService<ILoggerFactory>();
+            var logger = loggerFactory.CreateLogger(ScriptConstants.LogCategoryHostAuthentication);
+
+            string message = null;
+            switch (context.Exception)
+            {
+                case SecurityTokenInvalidIssuerException iex:
+                    message = $"Token issuer validation failed for issuer '{iex.InvalidIssuer}'.";
+                    break;
+                case SecurityTokenInvalidAudienceException iaex:
+                    message = $"Token audience validation failed for audience '{iaex.InvalidAudience}'.";
+                    break;
+                default:
+                    message = $"Token validation failed.";
+                    break;
+            }
+
+            logger.LogError(context.Exception, message);
```

Comment:
These security exceptions are safe to pass down to our logging layer because when formatted they mask/remove PII. Issuer and Audience aren't considered PII for our tokens so we log them above.